noxfile.py

```diff
@@ -223,11 +223,17 @@ def run_unit(session, install_test_extra):
 
 @nox.session(python=UNIT_TEST_PYTHON_VERSIONS)
 def unit(session):
+    session.install(
+        "scikit-learn>=1.2.2",
+    )
```

Comment:
If there is optional functionality that needs `scikit-learn`, please use the "extras" section in `setup.py`. Don't install packages this way.